source/io/module_javaScriptInvoke.js

```diff
@@ -516,7 +550,10 @@
 
             // assume synchronous invocation since the completion callback was never retrieved
             if (completionCallbackStatus.retrievalState === completionCallbackRetrievalEnum.AVAILABLE) {
-                tryUpdateReturnValue(functionName, returnTypeName, returnValuePointer, returnValue, errorStatusPointer, errorCodePointer, errorSourcePointer, completionCallbackStatus);
+                var errorStatus = Module.eggShell.dataReadBoolean(errorStatusPointer);
+                if (!errorStatus) {
```

Comment:
why was this case added?

So one thing we tried to do is represent the possible states of the jsli call using completionCallbackRetrievalEnum and completionCallbackInvocationEnum because there are a lot of possible configurations of the state of the JSLI function call.

What is the case where we are trying to look at Vireo's state of the error cluster? WE should try and capture it in the state enums or have the condition resolved sooner. IE if the input to the JSLI call is an error we should handle that much earlier.